mage/main.go

```diff
@@ -115,6 +118,7 @@ func Parse(stdout io.Writer, args []string) (inv Invocation, mageInit, showVersi
 	fs.BoolVar(&inv.List, "l", false, "list mage targets in this directory")
 	fs.BoolVar(&inv.Help, "h", false, "show this help")
 	fs.BoolVar(&mageInit, "init", false, "create a starting template if no mage files exist")
+	fs.IntVar(&inv.Timeout, "t", 0, "timeout in seconds, Default (0) no timeout")
```

Comment:
I think this is better as a DurationVar... it's nice to be able to do `-t 30s` or `-t 5m` etc

Even if we decide to keep the flag as an int, the Invocation Timeout field should definitely be a time.Duration, it makes testing easier, and it's the correct type for durations :)